geom/tests/Test_FieldBLAS.pf

```diff
@@ -0,0 +1,926 @@
+
+   subroutine test_FieldGetCptr()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldGetCptr'
+      type(ESMF_Field) :: x
+      type(c_ptr) :: cptr
+      integer :: status
+
+      RUNNING(TEST_NAME)
+
+      x = XR4
+      call FieldGetCptr(x, cptr, rc = status)
+      @assertTrue(SUCCESS(status), message='Unable to get c_ptr for rank-2 ESMF_KIND_R4 field')
+      
+      x = XR8
+      call FieldGetCptr(x, cptr, rc = status)
+      @assertTrue(SUCCESS(status), message='Unable to get c_ptr for rank-2 ESMF_KIND_R8 field')
+
+      PASSED(TEST_NAME)
+   end subroutine test_FieldGetCptr
+
+   @Test
+   !wdb fixme  Probably should test for non-conformable fields
+   subroutine test_FieldsAreConformableR4()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldsAreConformableR4'
+      type(ESMF_Field) :: x, y
+      integer :: status, rc
+      logical :: are_conformable
+
+      RUNNING(TEST_NAME)
+
+      x = XR4
+      y = YR4
+      are_conformable = .FALSE.
+      are_conformable = FieldsAreConformable(x, y, rc = status)
+      @assertTrue(are_conformable)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+
+      PASSED(TEST_NAME)
+
+   end subroutine test_FieldsAreConformableR4
+
+   !wdb fixme  Probably should test for non-conformable fields
+   @Test
+   subroutine test_FieldsAreConformableR8()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldsAreConformableR8'
+      type(ESMF_Field) :: x, y
+      integer :: status, rc
+      logical :: are_conformable
+
+      RUNNING(TEST_NAME)
+
+      x = XR8
+      y = YR8
+      are_conformable = .FALSE.
+      are_conformable = FieldsAreConformable(x, y, rc = status)
+      @assertTrue(are_conformable)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+
+      PASSED(TEST_NAME)
+
+   end subroutine test_FieldsAreConformableR8
+
+   @Test
+   !
+   subroutine test_FieldsAreSameTypeKind()
+      character(len=*), parameter :: TEST_NAME = 'test_FieldsAreSameTypeKind'
+      type(ESMF_Field) :: x, y
+      integer :: status, rc
+      logical :: are_same_typekind
+
+      RUNNING(TEST_NAME)
+
+      x = XR4
+      y = YR4
+      are_same_typekind = .FALSE.
+      are_same_typekind = FieldsAreSameTypeKind(x, y, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+      @assertTrue(are_same_typekind) 
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+
+      x = XR8
+      y = YR8
+      are_same_typekind = .FALSE.
+      are_same_typekind = FieldsAreSameTypeKind(x, y, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+      @assertTrue(are_same_typekind)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+
+      y = YR4
+      are_same_typekind = .TRUE.
+      are_same_typekind = FieldsAreSameTypeKind(x, y, rc = status)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
+      @assertFalse(are_same_typekind)
+      _ASSERT(SUCCESS(status), MSG('Call failed', status))
```

Comment:
Recently I've taken to introducing an `_RC` macro for the unit tests to avoid the tedious checking during unit tests and focus on the "meat".  Requires making an explicit pFUnit call rather than using the `@assert` mechanism.

OTOH, your approach of relying on MAPL's own error handling suggests that maybe even the regular macro would work this way.